src/props.ts

```diff
@@ -0,0 +1,219 @@
+import cx from "classnames";
+import { HTMLProps } from "react";
+
+export function kebabCaseToCamelCase(str: string) {
+  return str.replaceAll(/-[a-z]/g, (g) => g[1]?.toUpperCase() ?? "");
+}
+
+/**
+ * Converts a CSS style string to an object
+ * that can be passed to a React component's
+ * `style` prop.
+ */
+export function cssToStyles(css: string) {
+  const cssJson = `{"${css
+    .replace(/;? *$/, "")
+    .replace(/;+ */g, '","')
+    .replace(/: */g, '":"')}"}`;
+
+  const obj = JSON.parse(cssJson);
```

Comment:
Should we use the CSSOM for this?

```
const stylesheet = new CSSStyleSheet();
stylesheet.insertRule(`* { ${css} }`);
const style = stylesheet.cssRules[0].style;
```

Then, you should have a `CSSStyleDeclaration`. Iterating over that yields keys and you can get the property values and priorities.

Reply:
Oooo yes, that's awesome, I like that much better

Reply:
Oh hm. `cssRules` returns a `CSSRuleList`; each item is a `CSSRule`, but I don't see a way to get a `CSSStyleDeclaration` from that? Weirdly, MDN says that "`document.styleSheets[0].cssRules[0].style` returns a `CSSStyleDeclaration` object", but that doesn't seem to follow from the types and the rest of the documentation...

Reply:
`cssRules[0].style` returns a `CSSStyleDeclaration` in my browser.

Reply:
Hrmmmm. It works when I actually do it in the browser. So... something weird with the types, then. Maybe it's just that I need to discriminate on the `type` property. I'll figure it out!

Reply:
Ok cool. I ended up just casting it because:

(a) we're the one creating the rule, we know it's a style rule
(b) Apparently `CSSRule.type` is deprecated and that leaves me sort of baffled as to how I'm meant to detect the type anyway?

Do you think there are any performance concerns about doing this like... many many times during rendering? Should we just have one stylesheet, insert the rule, and then delete it again?

Reply:
Having a single stylesheet seems like a reasonable optimization. No idea how expensive any of this is.

Reply:
Don't worry about it, though. It looks like the constructor can run 4 million times a second on my machine.

Reply:
Hahah I was just setting up that same test. Great!